### PR TITLE
`addenv()`, by default, should inherit from `ENV`

### PR DESCRIPTION
**Pull request by staticfloat**
The spirit of `addenv()` is to be non-destructive, however because a
`Cmd` object with an `env` member that is set to `nothing` (the default)
inherts from the current environment when it is run.  This means that
the following, rather surprising behavior holds true on current master:

```
julia> ENV["FOO"] = "foo"
       run(`/bin/bash -c "echo \$FOO \$BAR"`)
       run(addenv(`/bin/bash -c "echo \$FOO \$BAR"`, "BAR" => "bar"))
foo
bar
```

This PR adds an `inherit` flag to `addenv()` to allow keeping this
behavior if it is actually desired (this might make sense if you are
constructing `.env` blocks over a series of `addenv()` calls and you
don't want to have to special-case your first operation as a `setenv()`
rather than an `addenv()`.

### CODE REVIEW

File: base/cmd.jl [hunk 1]
```diff
@@ -246,14 +246,19 @@ setenv(cmd::Cmd, env::Pair{<:AbstractString}...; dir="") =
 setenv(cmd::Cmd; dir="") = Cmd(cmd; dir=dir)
 
 """
-    addenv(command::Cmd, env...)
+    addenv(command::Cmd, env...; inherit::Bool = true)
 
 Merge new environment mappings into the given `Cmd` object, returning a new `Cmd` object.
-Duplicate keys are replaced.
+Duplicate keys are replaced.  If `cmd` does not contain any environment values set already,
```
Comment (Sacha0) [inline base/cmd.jl:252]: `cmd` -> `command` I think? :)

File: test/spawn.jl [hunk 2]
```diff
@@ -711,6 +711,28 @@ end
     @test strip(String(read(cmd))) == "bar bar"
     cmd = addenv(cmd, ["FOO=baz"])
     @test strip(String(read(cmd))) == "baz bar"
+
+    # Test that `addenv()` works properly with `inhert`
```
Comment (Sacha0) [inline test/spawn.jl:715]: `inhert` -> `inherit`? :)

**Review by Sacha0 — APPROVED** [review]
Looks great! :)

File: base/cmd.jl (hunk 1 above)
Comment (staticfloat) [inline base/cmd.jl:252]:
```suggestion
Duplicate keys are replaced.  If `command` does not contain any environment values set already,
```

File: test/spawn.jl (hunk 2 above)
Comment (staticfloat) [inline test/spawn.jl:715]:
```suggestion
    # Test that `addenv()` works properly with `inherit`
```

**Comment by KristofferC** [comment]
Feels like a feature so maybe shouldn't be backported? Or is there a specific reason why it should be?

**Comment by staticfloat** [comment]
It should be because `addenv()` was added in 1.6, and without this piece of functionality it's pretty broken, including in the cases that it's already been used in Base.

**Pull request merged**